client/verta/verta/_registry/modelversion.py

```diff
@@ -234,8 +234,11 @@ def log_model(self, model, custom_modules=None, model_api=None, artifacts=None,
         if artifacts:
             self.add_attribute(_MODEL_ARTIFACTS_ATTR_KEY, artifacts, overwrite=overwrite)
 
-        if isinstance(model, six.string_types):  # filepath
-            model = open(model, 'rb')
+        if isinstance(model, six.string_types):
+            if os.path.isdir(model):
+                model = _artifact_utils.zip_dir(model)
```

Comment:
curious how should we expect clients to consume a model built from a directory. it looks like the get model method will return bytes representing the zip file? do we expect the clients to unzip that as a follow up step?

Reply:
@houqp https://github.com/VertaAI/modeldb/pull/1973 will allow `run.download_artifact()` and `run.download_model()` to automatically unzip the artifact into a directory on download, which should fit this usecase.

`get_artifact()` and `get_model()` are usually instead used for direct Python objects, rather than files/directories.

Reply:
Ha, got it, sorry i missed that PR, thanks for the clarification 👍

Reply:
No worries—thank you for asking! Helps me make sure common workflows are fully supported.